overlord/devicestate/handlers_serial.go

```diff
@@ -679,6 +689,46 @@ func getSerialRequestConfig(t *state.Task, regCtx registrationContext, client *h
 	return &cfg, nil
 }
 
+func shouldRequestSerial(s *state.State, gadgetName string, deviceSerial string) (bool, error) {
+	tr := config.NewTransaction(s)
+
+	var storeAccess string
+	if err := tr.GetMaybe("core", "store.access", &storeAccess); err != nil {
+		return false, err
+	}
+
+	// if there isn't a gadget, just use store.access to determine if we should
+	// request
+	if gadgetName == "" {
+		return storeAccess != "offline", nil
+	}
+
+	var deviceServiceAccess string
+	if err := tr.GetMaybe(gadgetName, "device-service.access", &deviceServiceAccess); err != nil {
+		return false, err
+	}
+
+	// if we have a gadget and device-service.access is set to offline, then we
+	// will not request a serial
+	if deviceServiceAccess == "offline" {
+		return false, nil
+	}
+
+	var deviceServiceURL string
+	if err := tr.GetMaybe(gadgetName, "device-service.url", &deviceServiceURL); err != nil {
+		return false, err
+	}
+
+	// if we already have a device serial and we'd be using the fallback
+	// device-service.url (which is the store), then use store.access to
+	// determine if we should request
+	if deviceSerial != "" && deviceServiceURL == "" {
```

Comment:
I don't understand why it matters whether we have a serial or not, we shouldn't do requestes if the url is ""  and storreAccess is offline

Reply:
Was a misunderstanding on my part, fixed now. Thanks!